test/src/tests-store.ts

```diff
@@ -288,6 +293,202 @@ describe('Store', () => {
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            expect(store.getLastSeenTime(testMPID)).to.equal(null);
+        });
+
+        it('should return null if mpid is null', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            expect(store.getLastSeenTime(null)).to.equal(null);
+        });
+
+        it('should return the current time if mpid matches current user', () => {
+            const userSpy = sinon.stub(
+                window.mParticle.getInstance().Identity,
+                'getCurrentUser'
+            );
+            userSpy.returns({
+                getMPID: () => 'testMPID',
+            });
+
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            expect(store.getLastSeenTime(testMPID)).to.equal(now.getTime());
+        });
+    });
+
+    describe('#setLastSeenTime', () => {
+        it('should set the lastSeenTime in the store', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setLastSeenTime(testMPID, 12345);
+            expect(store.persistenceData[testMPID].lst).to.equal(12345);
+        });
+
+        it('should set the lastSeenTime in persistence', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setLastSeenTime(testMPID, 12345);
+            const fromPersistence = window.mParticle
+                .getInstance()
+                ._Persistence.getPersistence();
+
+            expect(fromPersistence[testMPID]).to.be.ok;
+            expect(fromPersistence[testMPID].lst).to.be.ok;
+            expect(fromPersistence[testMPID].lst).to.equal(12345);
+        });
+
+        it('should override persistence with store values', () => {
+            const persistenceValue = JSON.stringify({
+                gs: {
+                    sid: 'sid',
+                    les: new Date().getTime(),
+                },
+                testMPID: {
+                    lst: 12345,
+                },
+                cu: testMPID,
+            });
+
+            localStorage.setItem(workspaceCookieName, persistenceValue);
+
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            store.setLastSeenTime(testMPID, 54321);
+            const fromPersistence = window.mParticle
+                .getInstance()
+                ._Persistence.getPersistence();
+
+            expect(fromPersistence[testMPID]).to.be.ok;
+            expect(fromPersistence[testMPID].lst).to.be.ok;
+            expect(fromPersistence[testMPID].lst).to.equal(54321);
```

Comment:
same comment as above about testing that what is on the store is overriding persistence.